test/hotspot/jtreg/runtime/cds/appcds/SharedArchiveConsistency.java

```diff
@@ -146,201 +81,102 @@ private static long getRandomBetween(long start, long end) throws Exception {
             System.out.println("Region " + shared_region_name[region] + " is empty. Nothing to corrupt.");
             return false;
         }
-        System.out.println("Corrupt " + shared_region_name[region] + " section, start = " + region_start_offset
-                           + " (header_size + 0x" + Long.toHexString(region_start_offset-header_size) + ")");
-        long bytes_written = 0L;
-        while (bytes_written < used[region]) {
-            writeData(fc, region_start_offset + bytes_written, bbuf);
-            bbuf.clear();
-            bytes_written += 4096;
-        }
-        if (fc.isOpen()) {
-            fc.close();
+        System.out.println("Corrupt " + shared_region_name[region] + " section, start = " + regionStartOffset
+                           + " (header_size + 0x" + Long.toHexString(regionStartOffset - CDSArchiveUtils.fileHeaderSize()) + ")");
+        long bytesWritten = 0L;
+        while (bytesWritten < used[region]) {
+            CDSArchiveUtils.writeData(jsaFile, regionStartOffset + bytesWritten, buf);
+            bytesWritten += 4096;
         }
+
         return true;
     }
 
     public static void modifyJsaHeader(File jsaFile) throws Exception {
-        FileChannel fc = getFileChannel(jsaFile);
         // screw up header info
-        byte[] buf = new byte[getFileHeaderSize(fc)];
-        ByteBuffer bbuf = ByteBuffer.wrap(buf);
-        writeData(fc, 0L, bbuf);
-        if (fc.isOpen()) {
-            fc.close();
-        }
+        byte[] buf = new byte[CDSArchiveUtils.fileHeaderSize()];
+        System.out.println("CDSArchiveUtils.fileHeaderSize = " + CDSArchiveUtils.fileHeaderSize());
+        CDSArchiveUtils.writeData(jsaFile, 0, buf);
     }
 
     public static void modifyJvmIdent(File jsaFile) throws Exception {
-        FileChannel fc = getFileChannel(jsaFile);
-        int headerSize = getFileHeaderSize(fc);
-        System.out.println("    offset_jvm_ident " + offset_jvm_ident);
-        byte[] buf = new byte[256];
-        ByteBuffer bbuf = ByteBuffer.wrap(buf);
-        writeData(fc, (long)offset_jvm_ident, bbuf);
-        if (fc.isOpen()) {
-            fc.close();
-        }
+        System.out.println("    offset_jvm_ident " + CDSArchiveUtils.offsetJvmIdent);
+        byte[] buf = new String("Bad JDK 1.0001").getBytes();
+        CDSArchiveUtils.writeData(jsaFile, (long)CDSArchiveUtils.offsetJvmIdent, buf);
     }
 
     public static void modifyHeaderIntField(File jsaFile, long offset, int value) throws Exception {
-        FileChannel fc = getFileChannel(jsaFile);
-        int headerSize = getFileHeaderSize(fc);
         System.out.println("    offset " + offset);
-        byte[] buf = ByteBuffer.allocate(4).putInt(value).array();
-        ByteBuffer bbuf = ByteBuffer.wrap(buf);
-        writeData(fc, offset, bbuf);
-        if (fc.isOpen()) {
-            fc.close();
-        }
+        byte[] buf = { (byte)(value >> 24), (byte)(value >> 16), (byte)(value >> 8), (byte)(value)};
```

Comment:
Isn't this the same as the original code which does a `ByteBuffer.wrap()`?
If so, why changing it.

Reply:
I make CDSARchiveUtils take byte[] as input, not ByteBuffer. Here just simply convert int to byte[].